lib/promiscuous/publisher/operation/base.rb

```diff
@@ -132,26 +113,19 @@ def payloads
     end
   end
 
-  def publish_payloads_async(options={})
+  def publish_payloads(options={})
     unlock_all_locks and return if @operation_payloads.blank?
 
     exchange    = options[:exchange]  || Promiscuous::Config.publisher_exchange
     routing     = options[:routing]   || Promiscuous::Config.sync_all_routing
-    raise_error = options[:raise_error].present? ? options[:raise_error] : false
+    async       = !!options[:async]
 
     payloads.each do |payload|
-      begin
-        Promiscuous::AMQP.publish(:exchange => exchange.to_s,
-                                  :key => routing.to_s,
-                                  :payload => payload,
-                                  :on_confirm => method(:unlock_all_locks))
-      rescue Exception => e
```

Comment:
`AMQP.publish` handles this. It doesn't raise if its async. Otherwise it raises.